ios/PurchasesHybridCommon/PurchasesHybridCommon/CommonFunctionality.swift

```diff
@@ -654,3 +640,26 @@ private extension CommonFunctionality {
     }
 
 }
+
+// MARK: StoreKit 2 Observer Mode
+@objc public extension CommonFunctionality {
+
+    @available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, *)
+    @objc(handleObserverModeTransactionForProductID:completion:) 
+    static func handleObserverModeTransaction(productID: String, completion: (([String: Any]?, ErrorContainer?) -> Void)?) {
+        _ = Task<Void, Never> {
+            let result = await StoreKit.Transaction.latest(for: productID)
+            if let result = result {
+                do {
+                    let transaction = try await Self.sharedInstance.handleObserverModeTransaction(.success(result))
+                    completion?(transaction?.dictionary, nil)
+                } catch {
+                    completion?(nil, ErrorContainer(error: error, extraPayload: [:]))
+                }
+            } else {
+                completion?(nil, nil)
```

Comment:
We could either return an error here (but I couldn't find any existing error code we could reuse, and I'm not sure this warrants adding a new one?) or return nil but log a message. Thoughts?

Reply:
In what situations could this happen? (result being nil). 

I think ideally we return either success or error. Since it's not a success (unless I'm missing something) I think we should try to return an error... Could we return an unknown error or some store error with some extra info?